daemon/graphdriver/copy/copy.go

```diff
@@ -9,32 +9,94 @@ import (
 	"syscall"
 	"time"
 
+	"github.com/docker/docker/pkg/parsers/kernel"
+
+	"sync/atomic"
+
 	"github.com/docker/docker/pkg/pools"
 	"github.com/docker/docker/pkg/system"
 	rsystem "github.com/opencontainers/runc/libcontainer/system"
 	"golang.org/x/sys/unix"
 )
 
-type copyFlags int
+var copyFileRangeDisabled int32
+
+// The maximum amount of bytes Linux will splice
+const maxRwCount = 0x7ffff000
+
+// Mode indicates whether to use hardlink or copy content
+type Mode int
 
 const (
-	copyHardlink copyFlags = 1 << iota
+	// Content creates a new file, and copies the content of the file
+	Content Mode = iota
+	// Hardlink creates a new hardlink to the existing file
+	Hardlink
 )
 
-func copyRegular(srcPath, dstPath string, mode os.FileMode) error {
+func isCopyFileRangePosssible(srcDir, dstDir string) bool {
```

Comment:
I'm not sure why you're doing this check for two random directories. I would recommend doing this check *once* inside an `init()` function, and then just checking the global bool directly when deciding between the "legacy copy" and the "new copy".